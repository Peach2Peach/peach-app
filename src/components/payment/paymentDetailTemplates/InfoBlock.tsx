import { View } from 'react-native'
import { Text } from '../..'
import tw from '../../../styles/tailwind'
import i18n from '../../../utils/i18n'
import { CopyAble } from '../../ui'

type Props = ComponentProps & {
  value: string
  name?: string
  copyable?: boolean
  onInfoPress?: () => void
  disputeActive?: boolean
}

export const InfoBlock = ({ value, name, copyable, onInfoPress, disputeActive, style }: Props) => (
  <View style={[tw`flex-row`, style]}>
    <View style={tw`w-25`}>
      {!!name && <Text style={!disputeActive ? tw`text-black-2` : tw`text-error-light`}>{i18n(name)}</Text>}
    </View>
    <View key={'paymentDetails-' + name} style={tw`flex-row items-center flex-1`}>
      <Text
        onPress={!!onInfoPress ? onInfoPress : undefined}
        style={[tw`flex-wrap subtitle-1 leading-base`, !!disputeActive && tw`text-error-dark`]}
      >
        {value}
      </Text>
      {copyable && (
        <View style={tw`w-6 h-4 ml-2`}>
          <CopyAble value={value} color={!!disputeActive ? tw`text-error-main` : undefined} />
        </View>
      )}
    </View>
  </View>
)
