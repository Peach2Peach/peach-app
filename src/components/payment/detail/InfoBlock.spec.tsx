import { createRenderer } from 'react-test-renderer/shallow'
import tw from '../../../styles/tailwind'
import { InfoBlock } from './InfoBlock'

describe('InfoBlock', () => {
  const onInfoPressMock = jest.fn()

  it('should renders a default InfoBlock', () => {
    const renderer = createRenderer()
    renderer.render(<InfoBlock value="value" />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
  it('should renders a full InfoBlock with style', () => {
    const renderer = createRenderer()
    renderer.render(<InfoBlock style={tw`mt-[2px]`} name="name" value="value" copyable onInfoPress={onInfoPressMock} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
