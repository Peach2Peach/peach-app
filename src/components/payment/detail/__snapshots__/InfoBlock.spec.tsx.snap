// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBlock should renders a default InfoBlock 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "width": 100,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <PeachText
      style={
        {
          "flexWrap": "wrap",
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 24,
        }
      }
    >
      value
    </PeachText>
  </View>
</View>
`;

exports[`InfoBlock should renders a full InfoBlock with style 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "marginTop": 2,
      },
    ]
  }
>
  <View
    style={
      {
        "width": 100,
      }
    }
  >
    <PeachText
      style={
        {
          "color": "#7D675E",
        }
      }
    >
      name
    </PeachText>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": "0%",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <PeachText
      onPress={[MockFunction]}
      style={
        {
          "flexWrap": "wrap",
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 24,
        }
      }
    >
      value
    </PeachText>
    <View
      style={
        {
          "height": 16,
          "marginLeft": 8,
          "width": 24,
        }
      }
    >
      <CopyAble
        value="value"
      />
    </View>
  </View>
</View>
`;
